spec/PeriodicBackgroundSync-index.bs

```diff
@@ -145,6 +148,30 @@ Fetch requests within the [=periodicsync event=] while [=periodicsync event/in t
 ## History Leaking ## {#history-leaking}
 Fetch requests within the [=periodicsync event=] while [=periodicsync event/in the background=] may reveal something about the client's navigation history to middleboxes on networks different from the one used to create the [=periodic sync registration=]. For instance, the client might visit site https://example.com, which registers a [=periodicsync event=], but based on the implementation, might not fire until after the user has navigated away from the page and changed networks. Middleboxes on the new network may see the fetch requests that the [=periodicsync event=] makes. The fetch requests are HTTPS so the request contents will not be leaked but the destination of the fetch request and domain may be (via DNS lookups and IP address of the request). To prevent this leakage of browsing history, the user agent MAY choose to only fire [=periodicsync event|periodicsync events=] on the network the [=periodic sync registration=] was made on, with the understanding that it will reduce usability by not allowing synchronization opportunistically.
 
+Security Considerations {#security}
+====================================
+Since Periodic Background Sync is a [=service worker=] based API, [[service-workers#security-considerations]] apply.
+
+In addition, if there is a disparity between the intervals at which the web app can synchronize with its servers and those at which the user visits the site, the [=origin=] gets more power than the user is aware of or intends. This is because the [=origin=] is able to execute code more frequently than the user intends, from within the [=periodicsync event=] while [=periodicsync event/in the background=].
+User agents should therefore implement countermeasures to prevent such disparity.
```

Comment:
But isn't it the goal to be able to synchronize more often than the user visits? What we really want is countermeasures to prevent undue synchronization with an origin the user doesn't really want to synchronize with. But in the case of updates to my (e.g.) calendar, I definitely do want to synchronize with my calendar server as often as necessary.

Reply:
I think your calendar server should send push messages when a new meeting is coming up, rather than sync'ing constantly, although that doesn't deal well with periods of being offline.

Reply:
If the user visits often anyway, the site can fetch content when the user is on the page. The case we're trying to solve here is the user visits the site during their commute, and it would've been nice to have synced beforehand.

It's also not necessary to sync as often as the user visits, only if the rate at which new content is released is also high.

Reply:
Also, Periodic Background Sync is not a good API for your calendar PWA which needs to fire events/show notifications at exact times because it won't run a task or show a notification in Doze mode. Scheduled notification triggers would fit the calendar app best, because it allows showing notifications at exact times.